homeassistant/components/ring/__init__.py

```diff
@@ -53,42 +42,14 @@ def token_updater(token):
     )
     ring = ring_doorbell.Ring(auth)
 
-    try:
-        await hass.async_add_executor_job(ring.update_data)
-    except ring_doorbell.AuthenticationError as err:
-        _LOGGER.warning("Ring access token is no longer valid, need to re-authenticate")
-        raise ConfigEntryAuthFailed(err) from err
+    coordinator = RingDataCoordinator(hass, ring)
+    await coordinator.async_config_entry_first_refresh()
 
     hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {
         RING_API: ring,
         RING_DEVICES: ring.devices(),
-        RING_DEVICES_COORDINATOR: GlobalDataUpdater(
-            hass, "device", entry, ring, "update_devices", DEVICES_SCAN_INTERVAL
-        ),
-        RING_NOTIFICATIONS_COORDINATOR: GlobalDataUpdater(
-            hass,
-            "active dings",
-            entry,
-            ring,
-            "update_dings",
-            NOTIFICATIONS_SCAN_INTERVAL,
-        ),
-        RING_HISTORY_COORDINATOR: DeviceDataUpdater(
-            hass,
-            "history",
-            entry,
-            ring,
-            lambda device: device.history(limit=10),
-            HISTORY_SCAN_INTERVAL,
-        ),
-        RING_HEALTH_COORDINATOR: DeviceDataUpdater(
-            hass,
-            "health",
-            entry,
-            ring,
-            lambda device: device.update_health_data(),
-            HEALTH_SCAN_INTERVAL,
-        ),
+        RING_DEVICES_COORDINATOR: coordinator,
+        RING_NOTIFICATIONS_COORDINATOR: RingNotificationsCoordinator(hass, ring),
```

Comment:
Why aren't we calling `async_config_entry_first_refresh()` on this coordinator?

We should call it for each coordinator

Reply:
I think it was because it refreshes every 5 seconds I'd decided not to, but I now see there's extra logic in the `async_config_entry_first_refresh`.  I've updated it to call that and it all works ok.